### PR TITLE
RALibRetro support

### PR DESCRIPTION
**Pull request by Jamiras**
* adds emulator enum RA_Libretro
* adds console enums provided by Scott in #web-dev
* replaces full window InvalidateRect calls with per-list item InvalidateRect calls to reduce flickering of toolkit windows
* renders overlay in a back buffer to reduce flickering of main window

### CODE REVIEW

File: RA_Integration/RA_Dlg_Memory.cpp [hunk 1]
```diff
@@ -793,51 +800,6 @@ INT_PTR Dlg_Memory::MemoryProc( HWND hDlg, UINT nMsg, WPARAM wParam, LPARAM lPar
 {
 	switch ( nMsg )
 	{
-		case WM_TIMER:
```
Comment (Jamiras) [inline RA_Integration/RA_Dlg_Memory.cpp]: timer was eliminated, search results are updated each time the memory viewer is updated

**Comment by meleu** [comment]
@Jamiras do you have a compiled version so I can test it?

**Comment by Jamiras** [comment]
The compiled version was uploaded to discord on Sunday. I believe it's in the repository now. The dll should be version "0.070-dlg_flicker". I held off creating the PR to do some more testing of my own.

**Comment by meleu** [comment]
@RetroAchievements/rasuite we are currently using on RALibretro the Integration built by Jamiras from the code of this PR.

I was talking with him on discord and he says he's confident about this PR, and from a user point of view it's working pretty fine.

Do you guys, more experienced with the codebase, have something to comment about the code?

The main benefit to get this merged (and then published by Scott as the official version) is that we won't need to have a different integration dll for RALibretro.

**Pull request merged**